components/analytics-tools/org.wso2.carbon.analytics.tools.backup/src/main/java/org/wso2/carbon/analytics/tools/backup/AnalyticsDataBackupTool.java

```diff
@@ -320,5 +336,170 @@ private static AnalyticsSchema readTableSchema(String basePath) throws IOExcepti
             }
         }
     }
-    
+
+    /**
+     * Backs up the file system to Local
+     *
+     * @param analyticsFileSystem
+     * @param tenantId
+     * @param baseDir
+     * @throws IOException
+     */
+    private static void backupFileSystem(AnalyticsFileSystem analyticsFileSystem, int tenantId,
+                                         File baseDir) throws IOException {
+        System.out.println("Backing up the filesystem to: " + baseDir);
+        backupFileSystemToLocal(analyticsFileSystem, "/", baseDir.getAbsolutePath());
+    }
+
+    /**
+     * Backing up the filesystem to the local recursively
+     * @param analyticsFileSystem
+     * @param path
+     * @param baseDir
+     * @throws IOException
+     */
+    private static void backupFileSystemToLocal(AnalyticsFileSystem analyticsFileSystem,
+                                                String path, String baseDir) throws IOException {
+        List<String> nodeList = analyticsFileSystem.list(path);
+        String parent_path = (path.equals("/")) ? path : path + File.separator;
+        String node_path = "";
+
+        for (String node : nodeList) {
+            node_path = parent_path + node;
+            if (analyticsFileSystem.length(node_path) == 0) { // the node is a directory
+                createDirectoryInLocalSystem(baseDir + File.separator + node_path);
+                backupFileSystemToLocal(analyticsFileSystem, node_path, baseDir);
+
+            } else {                                          // the node is a file
+                AnalyticsFileSystem.DataInput input = analyticsFileSystem.createInput(node_path);
+                byte[] dataInBuffer = new byte[READ_BUFFER_SIZE];
+                int i;
+                FileOutputStream out = new FileOutputStream(baseDir+node_path);
+                while ((i = input.read(dataInBuffer, 0, dataInBuffer.length)) > 0) {
+                    out.write(dataInBuffer);
```

Comment:
This is a bug, the variable "i" is not used, which should be passed into the "write" method. Always check the IDE warnings, which would have been signaled by it.